src/providers/Auth/AuthProvider.tsx

```diff
@@ -1,96 +1,169 @@
-      try {
-        await setPersistence(auth, browserSessionPersistence);
-        console.log("Persistence set to local");
-      } catch (error) {
-        console.error("Failed to set persistence:", error);
-      }
-    };
-
-    setAuthPersistence();
-
-    const fetchUserData = async (user: User) => {
-      const response = await fetch(
-        `${import.meta.env.VITE_API_URL}/api/v1/profile/${user.uid}`,
-        {
-          method: "GET",
-          headers: {
-            "Content-Type": "application/json",
-          },
-        }
-      );
-
-      if (!response.ok) {
-        throw new Error("Failed to fetch user data");
-      }
-      const data: userDocument = await response.json();
-      setUserData(data);
-    };
-
-    const unsubscribe = auth.onAuthStateChanged((user) => {
-      setCurrentUser(user);
-      if (user) {
-        try {
-          fetchUserData(user);
-        } catch (e) {
-          console.error("Failed to fetch user data: ", e);
-        }
-      }
-      setIsLoading(false);
-    });
-
-    return () => unsubscribe();
-  }, [auth]);
-
-  useEffect(() => {
-    if (!FF.stripePayment) {
-      setIsSignedIn(!!currentUser && !!userData && userData.paymentVerified!);
-    } else {
-      setIsSignedIn(!!currentUser && !!userData);
-    }
-  }, [currentUser, userData]);
-
-  const logout = () => {
-    setIsSignedIn(false);
-    return auth.signOut();
-  };
-
-  const value: AuthContextType = {
-    currentUser,
-    userData,
-    setUserData,
-    logout,
-    isSignedIn,
-  };
-
-  return (
-    <AuthContext.Provider value={value}>
-      {!isLoading && children}
-    </AuthContext.Provider>
-  );
-};
+import {useContext, useEffect, useState} from "react";
+import {AuthContextType, AuthProviderProps} from "./types";
+import {onboardingBody, userDocument} from "../../types/api";
+import AuthContext from "./AuthContext";
+import {useAuth0} from "@auth0/auth0-react";
+import FF from "../../../feature-flag.json";
+
+export const useAuth = (): AuthContextType => {
+    const context = useContext(AuthContext);
+    if (!context) {
+        throw new Error("useAuth must be used within an AuthProvider");
+    }
+    return context;
+};
+
+export function AuthProvider({children}: AuthProviderProps) {
+    const {user, isAuthenticated, getIdTokenClaims} = useAuth0();
+    const [userData, setUserData] = useState<userDocument | null>(null);
+    const [isSignedIn, setIsSignedIn] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+        const handleLogin = async () => {
+            setIsLoading(true);
+            try {
+                await migrateOldUser();
+
+                const data = await fetchUserData(user!.sub!);
+                if (data) {
+                    setUserData(data);
+                }
+
+                if (!FF.stripePayment) {
+                    setIsSignedIn(!!user && !!userData && userData.paymentVerified!);
```

Comment:
userData is null initially, so let's use data instead of userData for the isSignedIn logic